src/main/java/io/cdap/plugin/http/source/common/HttpErrorHandler.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.http.source.common;
+
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * A class which calculates an error handling strategy for http status codes.
+ */
+public class HttpErrorHandler {
+  private Map<Pattern, HttpErrorHandlingStrategy> httpErrorsHandlingMap;
+
+  public HttpErrorHandler(BaseHttpSourceConfig config) {
+    this.httpErrorsHandlingMap = config.getHttpErrorsHandlingMap();
+  }
+
+  public HttpErrorHandlingStrategy getErrorHandlingStrategy(int httpCode) {
+    String httpCodeString = Integer.toString(httpCode);
+
+    for (Map.Entry<Pattern, HttpErrorHandlingStrategy> entry : httpErrorsHandlingMap.entrySet()) {
+      Matcher matcher = entry.getKey().matcher(httpCodeString);
+      if (matcher.matches()) {
+        return entry.getValue();
+      }
+    }
+
+    throw new IllegalStateException(String.format("No error handling strategy defined for HTTP status code '%d'. " +
```

Comment:
Think it's better to have a default instead of requiring that every status code possibility is covered. Perhaps failing should be the default?

In both cases the pipeline will fail, but we want it to fail with a message about the HTTP error, not with this error.

Reply:
We already have a default in the widget.
`"default": "2..:Success,.*:Fail"`

Reply:
Yes but somebody can remove those defaults, or the pipeline can be created through the REST API.

Reply:
Done